src/main/java/com/google/sps/servlets/ServletConfig.java

```diff
@@ -0,0 +1,32 @@
+package com.google.sps.servlets;
+
+import com.google.sps.servlets.CareerQuizServlet;
+import com.google.sps.data.CareerQuestionDatabase;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.servlet.GuiceServletContextListener;
+import com.google.inject.servlet.ServletModule;
+
+/** Sets up servlets for the web application and injects necessary dependencies*/
+public class ServletConfig extends GuiceServletContextListener {
+  @Override
+  protected Injector getInjector() {
+    return Guice.createInjector(new ServletModule() {
+      @Override
+      protected void configureServlets() {
+        super.configureServlets();
+
+        serve("/careerquiz").with(CareerQuizServlet.class);
+                
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        CareerQuestionDatabase careerQuestionDatabase = new CareerQuestionDatabase(datastore);
+
+        // define above specific instance to be injected into CareerQuizServlet whenever a 
+        // CareerQuestionDatabase is requested
+        bind(CareerQuestionDatabase.class).toInstance(careerQuestionDatabase); 
+      }
+    });
```

Comment:
You could make a new java file like `DatabaseServletModule.java`: where it would just be a class like:

```
public class DatabaseServletModule extends ServletModule {
    @Override
    protected void configureServlets() {
        super.configureServlets();

        serve("/careerquiz").with(CareerQuizServlet.class);

        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
        CareerQuestionDatabase careerQuestionDatabase = new CareerQuestionDatabase(datastore);

        // define above specific instance to be injected into CareerQuizServlet whenever a 
        // CareerQuestionDatabase is requested
        bind(CareerQuestionDatabase.class).toInstance(careerQuestionDatabase); 
    } 
}
```

Then you can do:
```
  protected Injector getInjector() {
    return Guice.createInjector(new DatabaseServletModule());
  }
```

Reply:
Great suggestion, done.